### PR TITLE
Recommend updating traitlets to 5.1.1 if jupyter fails to start

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #8295

image

### CODE REVIEW

File: src/client/common/utils/localize.ts [hunk 1]
```diff
@@ -379,6 +379,18 @@ export namespace DataScience {
...
+        'DataScience.failedToStartJupyter',
+        "Failed to start Jupyter in the environment '{0}'. \nView Jupyter [log](command:jupyter.viewOutput) for further details."
+    );
+    export const failedToStartJupyterWithErrorInfo = localize(
+        'DataScience.failedToStartJupyterWithErrorInfo',
+        "Failed to start Jupyter in the environment '{0}'. \n{1} \nView Jupyter [log](command:jupyter.viewOutput) for further details."
+    );
+    export const failedToStartJupyterDueToOutdatedTraitlets = localize(
+        'DataScience.failedToStartJupyterDueToOutdatedTraitlets',
+        "Failed to start Jupyter in the environment '{0}' possibly due to an outdated version of 'traitlets'. \n{1} \nConsider updating the 'traitlets' module to '5.1.1' or later. \nView Jupyter [log](command:jupyter.viewOutput) for further details."
```
Comment (DonJayamanne) [inline src/client/common/utils/localize.ts:392]: 3 variations of errors we display.
If we have an error message from python (e.g. we have a python stack trace & have been able to capture the python error, then we display that, else we display a generic error message).
If we can find a specific error thats generally associated with the traitlets being old, then we have another for that.

File: src/datascience-ui/error-renderer/index.ts [hunk 2]
```diff
@@ -96,6 +97,11 @@ export const activate: ActivationFunction = (_context) => {
                         line = line.replace(matches[0], `<a href='command:${matches[1]}'>${matches[2]}</a>`);
                     }
                 }
+                while ((matches = akaMsLinks.exec(line)) !== null) {
```
Comment (DonJayamanne) [inline src/datascience-ui/error-renderer/index.ts:100]: Modified to support https links to aka.ms links
This way the `More info` is also displayed in the error output

File: src/client/datascience/errors/errorHandler.ts [hunk 3]
```diff
@@ -219,15 +227,78 @@ export class DataScienceErrorHandler implements IDataScienceErrorHandler {
...
+                  this.workspace.workspaceFolders || []
+              )})`
+            : '';
+        if (
+            jupyterInterpreter &&
+            failureInfoFromMessage?.reason === KernelFailureReason.jupyterStartFailureOutdatedTraitlets
+        ) {
+            void this.showMessageWithMoreInfo(
+                DataScience.failedToStartJupyterDueToOutdatedTraitlets().format(envDisplayName, pythonError || ''),
+                'https://aka.ms/kernelFailuresJupyterTrailtletsOutdated',
```
Comment (IanMatthewHuff) [inline src/client/datascience/errors/errorHandler.ts:292]: Looks like a misspell here, or is the aka link also misspelled?

**Review by IanMatthewHuff — APPROVED** [review]
LGTM with possible spelling fix

File: src/client/datascience/errors/errorHandler.ts (hunk 3 above)
Comment (DonJayamanne) [reply src/client/datascience/errors/errorHandler.ts:292]: Yuck, i'll leave that spelling, as the link works and from what I know we can't delete those links.
Also no harm in that.

**Review by rchiodo — APPROVED** [review]

**Pull request merged**